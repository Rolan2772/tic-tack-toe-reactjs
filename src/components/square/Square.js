import React from "react";
import Button from '@material-ui/core/Button';
import {withStyles} from '@material-ui/core/styles';

const styles = {
    cell: {
        'min-height': '36px'
    }
};

export const Square = props => {
    return (
        <Button variant="outlined" onClick={props.onClick} className={props.classes.cell}>
            {props.value || ''}
        </Button>
    )
};

export default withStyles(styles)(Square);