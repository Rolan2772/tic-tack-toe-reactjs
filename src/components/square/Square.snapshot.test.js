import React from 'react';
import Square from './Square';
import renderer from 'react-test-renderer';

describe('<Square />', () => {

    test('should match snapshot', () => {
        const component = renderer.create(
            <Square value="X" onClick={() => ''}/>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('should match snapshot after click', () => {
        const component = renderer.create(
            <Square value="X" onClick={() => ''}/>
        );

        let tree = component.toJSON();
        tree.props.onClick();
        tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});

