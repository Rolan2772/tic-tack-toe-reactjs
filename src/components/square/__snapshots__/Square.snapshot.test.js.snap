// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> should match snapshot after click 1`] = `
<button
  className="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-outlined-10 Square-cell-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    X
  </span>
  <span
    className="MuiTouchRipple-root-31"
  />
</button>
`;

exports[`<Square /> should match snapshot with empty value 1`] = `
<button
  className="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-outlined-10 Square-cell-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    X
  </span>
  <span
    className="MuiTouchRipple-root-31"
  />
</button>
`;

exports[`<Square /> should match snapshot with value 1`] = `
<button
  className="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-outlined-10 Square-cell-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    X
  </span>
  <span
    className="MuiTouchRipple-root-31"
  />
</button>
`;
